lib/rubygem_fs.rb

```diff
@@ -0,0 +1,78 @@
+module RubygemFs
+
+  def self.instance
+    @fs ||=
+      if Rails.env.development?
+        RubygemFs::Local.new
+      else
+        RubygemFs::S3.new
+      end
+  end
+
+  def self.mock!
+    @fs = RubygemFs::Local.new.tap do |fs|
+      def fs.base_dir
+        @dir ||= Dir.mktmpdir
+      end
+    end
+  end
+
+  def self.s3!(host)
+    uri = URI(host)
+    @fs = RubygemFs::S3.new
+    s3 = @fs.s3(access_key_id: 'k', secret_access_key: 's',  proxy: {host: uri.host, port: uri.port})
```

Comment:
OK. Meant to ask - does this still allow pushing of gems locally to unicorn
(and pow, if we ever get that working again?)

On Fri, Apr 3, 2015 at 10:52 AM, Arthur Nogueira Neves <
notifications@github.com> wrote:

> In lib/rubygem_fs.rb
> <https://github.com/rubygems/rubygems.org/pull/925#discussion_r27736665>:
>
> > +        RubygemFs::S3.new
> > +      end
> > +  end
> > +
> > +  def self.mock!
> > +    @fs = RubygemFs::Local.new.tap do |fs|
> > +      def fs.base_dir
> > +        @dir ||= Dir.mktmpdir
> > +      end
> > +    end
> > +  end
> > +
> > +  def self.s3!(host)
> > +    uri = URI(host)
> > +    @fs = RubygemFs::S3.new
> > +    s3 = @fs.s3(access_key_id: 'k', secret_access_key: 's',  proxy: {host: uri.host, port: uri.port})
>
> this s3! method is only used when you pass a ENV var S3_PROXY . Thats so
> we can simulate using fake-s3 in local development. The key and secret are
> just mocks on fake-s3.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rubygems/rubygems.org/pull/925/files#r27736665>.
>